Mathlib/Order/PrimeSeparator.lean

```diff
@@ -63,7 +63,8 @@ theorem prime_ideal_of_disjoint_filter_ideal (hFI : Disjoint (F : Set α) (I : S
             ⟨le_trans (hcS hJ).2.1 (le_sSup hJ), fun J hJ ↦ (hcS hJ).2.2⟩⟩
 
   -- Thus, by Zorn's lemma, we can pick a maximal ideal J in S.
-  obtain ⟨Jset, ⟨Jidl, IJ, JF⟩, ⟨_, Jmax⟩⟩ := zorn_subset_nonempty S chainub I IinS
+  obtain ⟨Jset, _, hmax⟩ := zorn_subset_nonempty S chainub I IinS
+  obtain ⟨Jidl, IJ, JF⟩ := hmax.prop
```

Comment:
I see this pattern a lot in this PR. Should we just have an additional lemma besides `zorn_subset_nonempty` that does the second step deconstruction too?